### PR TITLE
Remove pointless `ENABLE_ACLK`/`ENABLE_CLOUD` split.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

The distinction is a holdover from years ago when we had multiple versions of the agent-cloud link in our code. These days, everything in the code cares only about `ENABLE_ACLK`, and `ENABLE_CLOUD` is just used in the claiming script.

This PR:

- Removes the `ENABLE_CLOUD` check in the claiming script so that it looks at only the things that the agent code looks at to determine cloud support.
- Removes the current `ENABLE_CLOUD` option and define as it now does nothing.
- Renames the existing `ENABLE_ACLK` CMake option to `ENABLE_CLOUD`, because that is properly descriptive of what the option actually does.

##### Test Plan

Local testing is required for this PR.

### CODE REVIEW

**Review by thiagoftsm — DISMISSED** [review]
This PR is also working as expected. I have normal access to cloud, LGTM!

**Comment by Ferroin** [comment]
Rebased to pick up latest changes, also added a bit more cleanup.

**Review by thiagoftsm — DISMISSED** [review]
PR is working as expected after last commits!

File: CMakeLists.txt
Comment (vkalintiris) [inline CMakeLists.txt:158]: Why do we need this and simply not use `ENABLE_CLOUD` throughout?

Comment (Ferroin) [reply CMakeLists.txt:158]: My intent here was to avoid changing C code in this PR (we use `ENABLE_ACLK` in essentially all places in the C code at the moment) because changing it would have made the PR massive and much harder to review, and thus theoretically would have delayed getting it merged. I plan to do a followup to convert the C code once this is merged, which will get rid of this bit of the CMakeLists file as well.

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts.

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts.

**Comment by Ferroin** [comment]
Closing as this is no longer relevant.

**Pull request closed without merging**
